homeassistant/components/ecobee/climate.py

```diff
@@ -651,7 +653,7 @@ def set_temperature(self, **kwargs):
 
     def set_humidity(self, humidity):
         """Set the humidity level."""
-        self.data.ecobee.set_humidity(self.thermostat_index, humidity)
+        self.data.ecobee.set_humidity(self.thermostat_index, int(humidity))
```

Comment:
Are only integers accepted and not floats?

Reply:
It actually get typecast as a string when passed to Ecobee servers.  The climate component converts the set_humidity value to a float and Ecobee API rejects the call if decimal points are in the string passed to it, so it is cast as an int to ensure no decimals are sent when making the call.

Reply:
Ok. :+1: